lib/activity_runtime_api.rb

```diff
@@ -448,4 +451,15 @@ def self.remove_report_embeddable_filters(external_activity)
     filters = template.offerings.map { |offering| offering.report_embeddable_filter }.compact
     filters.each { |filter| filter.clear }
   end
+
+
+  def self.update_external_report(external_activity, report_url)
+    external_report = nil
+    if report_url
+      external_report = ExternalReport.find_by_url(report_url)
```

Comment:
This is going to abort the publish or update if the external report isn't defined in the portal.  Is that the desired result?